### DISCUSSION TITLE
Passing data to your webhook

### DISCUSSION THREAD

**Discussion by TannerGabriel** [argoproj-labs/argocd-notifications#336] [Q&A]
How can I add custom dynamic information to my webhook e.g. an `id` needs to be set whenever sending the webhook, which defines the object the webhook is meant for.

Here is an example:

```
template.keptn-deployment-started: |
    webhook:
      keptn:
        method: POST
        path: /api/controlPlane/v1/event
        body: |
          {
            "type": "sh.keptn.event.deployment.started",
            "contenttype": "application/json",
            "specversion": "0.2",
            "source": "test-event",
            "id": "9021830578932789191",
            "shkeptncontext": "08735340-6f9e-4b32-97ff-3b6c292bc112",
            "data": {
              "project": "sockshop",
              "stage": "dev",
              "service": "carts",
              "testStrategy": "{{app.operation.info}}",
              "deploymentStrategy": "direct",
              "tag": "0.10.1",
              "image": "docker.io/keptnexamples/carts",
              "labels": {
                "testid": "12345",
                "buildId": "build17",
                "runby": "JohnDoe"
              },
              "deploymentURILocal": "http://carts.sockshop-staging.svc.cluster.local",
              "deploymentURIPublic":  "https://carts.sockshop-staging.my-domain.com"
            }
          }
```

In the example above all the information is hardcoded, but parameters like the `id` and `shkeptncontext` would need to be dynamically inserted and therefore be different for every sync.

**Comment by ryota-sakamoto** [comment]
I'm not sure what kind of values you want to set it in json.
You can add values in template like `{{ .app.metadata.name }}` from Application Resource.
https://github.com/argoproj-labs/argocd-notifications/blob/master/catalog/templates/app-deployed.yaml#L8

**Comment by TannerGabriel** [comment]
@ryota-sakamoto Thank you for the answer. I already know that you can get information from the application resources into your webhook using the above-outlined syntax. My problem is automatically getting custom information like a request-specific id or context into Argo so I can use it. This information is different for every sync end therefore for every webhook. So here is what I would need to do:

1. Get the information into the sync process
2. Use the information inside of the webhook

Hopefully, this describes my current situation better. So is there some parameter or option to get information into the sync process?
